src/training_hub/jit_checkpoint.py

```
@@ -0,0 +1,151 @@
+        self._handle_preemption(context)
+
+    def on_save(self, context: TrainingHubContext) -> None:
+        if not context.output_dir or context.step <= 0:
+            return
+
+        if context.is_main_process:
+            mark_checkpoint_complete(context.output_dir, context.step)
+
+        checkpoint_path = context.metrics.get("checkpoint_path")
+        if checkpoint_path and context.is_main_process:
+            enqueue_checkpoint_upload(checkpoint_path)
```

Ruff (0.16.3)

[error] Undefined name `enqueue_checkpoint_upload`

(F821)